custom_components/ha_washdata/__init__.py

```
@@ -646,6 +646,19 @@ async def handle_trim_cycle(call: ServiceCall) -> None:
     await store_account.async_load(hass)  # integration-wide online flag + account
+
+    # Cache the integration version from HA's already-loaded manifest (no file IO).
+    # ws_get_constants reads it from hass.data; we can't do a module-level read_text
+    # in ws_api.py because the module is imported lazily inside this coroutine and the
+    # IO runs on the event loop (#328/#335).
+    if "ha_washdata_version" not in hass.data:
+        try:
+            from homeassistant.loader import async_get_integration as _aget_integration  # pylint: disable=import-outside-toplevel
+            _integ = await _aget_integration(hass, DOMAIN)
+            hass.data["ha_washdata_version"] = _integ.manifest.get("version", "") or ""
+        except Exception:  # pylint: disable=broad-exception-caught
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)